interactive_engine/executor/engine/pegasus/pegasus/src/progress.rs

```diff
@@ -269,7 +269,10 @@ impl EndOfScope {
 
     pub(crate) fn contains_source(&self, src: u32) -> bool {
         (self.tag.is_root() && src == 0)
-            || (!self.tag.is_root() && self.peers.value() > 0 && self.peers.contains_source(src))
+            || (!self.tag.is_root()
```

Comment:
Consider using `if .. else` to simplify the condition check, like
```
if self.tag.is_root() {
    src == 0  
} else {
    // index: get the worker for handling the current subtask (add a function) 
    if src != index {
       (self.peers.value() > 0 && self.peers.contains_source(src)))
    } else {
        true 
    }
}
```

In addition, please properly comment this function to explain the given cases.